### PR TITLE
DAOS-15702 dtx: metrics for sync DTX_COMMIT

### PR DESCRIPTION
**Pull request by Nasf-Fan**
The metrics for sync DTX_COMMIT will help us to well understand the average degree of DTX batched commit.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/dtx/dtx_srv.c [hunk 1]
```diff
@@ -114,6 +114,13 @@ dtx_metrics_alloc(const char *path, int tgt_id)
 			D_WARN("Failed to create DTX RPC cnt metric for %s: "
 			       DF_RC"\n", dtx_opc_to_str(opc), DP_RC(rc));
 	}
+
+	rc = d_tm_add_metric(&metrics->dpm_total[DTX_PROTO_SRV_RPC_COUNT], D_TM_COUNTER,
+			     "total number of processed sync DTX_COMMIT", "ops",
+			     "%s/ops/sync_dtx_commit/tgt_%u", path, tgt_id);
```
Comment (johannlombardi) [inline src/dtx/dtx_srv.c:120]: Looks good. So if i want to know how many piggy-backed commit notifications happened, i will be able to calculate it from this metrics + existing ones?

**Review by johannlombardi — DISMISSED** [review]

File: src/dtx/dtx_srv.c (hunk 1 above)
Comment (Nasf-Fan) [reply src/dtx/dtx_srv.c:120]: This metrics stands for how many sync DTX commit (RPC) arrived from other DTX leaders.

Assume that:
1. The batched commit RPC (from others to current target) count is A.
2. The average bathed commit (from others to current target) degree is D.
3. The sync_dtx_commit (from others to current target) count is S.
4. The total DTX count on current target is T.
5. The count of DTX which leader resides on current target is L.

The DTX count that are committed via piggyback is equal to: T - L - (A * D + S). Unfortunately, we do not have metrics for "L", that contains both batched_commit and sync_commit from this target to others. If you want, I can add.

**Review by liuxuezhao — APPROVED** [review]

File: src/dtx/dtx_srv.c (hunk 1 above)
Comment (liuxuezhao) [reply src/dtx/dtx_srv.c:120]: maybe can add it to comment somewhere, I guess only you understand how to calculate it :)

Comment (Nasf-Fan) [reply src/dtx/dtx_srv.c:120]: OK, I will add comment in subsequent DTX related patch.

**Review by johannlombardi — APPROVED** [review]

**Comment by Nasf-Fan** [comment]
Ping @daos-stack/daos-gatekeeper , thanks!

**Review by gnailzenh — APPROVED** [review]

**Pull request merged**